atc/atccmd/command.go

```diff
@@ -500,13 +500,16 @@ func (cmd *RunCommand) constructAPIMembers(
 		return nil, err
 	}
 
+	resourceFactory := resource.NewResourceFactory()
 	dbResourceCacheFactory := db.NewResourceCacheFactory(dbConn, lockFactory)
-	resourceFetcherFactory := resource.NewFetcherFactory(lockFactory, clock.NewClock(), dbResourceCacheFactory)
+	fetchSourceFactory := resource.NewFetchSourceFactory(dbResourceCacheFactory, resourceFactory)
```

Comment:
Looks like the behavior in `FetchSource` could be collapsed into `Fetcher`:
- we only ever instantiate them in `atccmd`
- `FetchSource`'s interface is only relied on by `Fetcher`

That would be.. 
![SO FETCH](https://media.giphy.com/media/bzASUU4JoBBny/giphy.gif)

Reply:
Might also be feasible to roll all of this up into the current resource factory, as its only there as a boundary for tests.. but I wouldn't block merging this on that change!